src/IR/IR.jl

```diff
@@ -760,6 +760,7 @@ end
 
 MModule(loc::Location=Location()) =
     MModule(API.mlirModuleCreateEmpty(loc))
+MModule(op::Operation) = MModule(API.mlirModuleFromOperation(op))
```

Comment:
I think we need this to prevent triggering both moduleDestroy and operationDestroy

```suggestion
MModule(op::Operation) = MModule(API.mlirModuleFromOperation(lose_ownership!(op)))
```